src/Halogen/HTML/Elements/Indexed.purs

```diff
@@ -0,0 +1,497 @@
+module Halogen.HTML.Elements.Indexed
+  ( Properties()
+  , Node()
+  , Leaf()
+  , a
+  , abbr
+  , acronym
+  , address
+  , applet
+  , area
+  , article
+  , aside
+  , audio
+  , b
+  , base
+  , basefont
+  , bdi
+  , bdo
+  , big
+  , blockquote
+  , body
+  , br
+  , button
+  , canvas
+  , caption
+  , center
+  , cite
+  , code
+  , col
+  , colgroup
+  , command
+  , datalist
+  , dd
+  , del
+  , details
+  , dfn
+  , dialog
+  , dir
+  , div
+  , dl
+  , dt
+  , em
+  , embed
+  , fieldset
+  , figcaption
+  , figure
+  , footer
+  , form
+  , h1
+  , h2
+  , h3
+  , h4
+  , h5
+  , h6
+  , head
+  , header
+  , hr
+  , html
+  , i
+  , iframe
+  , img
+  , input
+  , ins
+  , kbd
+  , keygen
+  , label
+  , legend
+  , li
+  , link
+  , main
+  , map
+  , mark
+  , menu
+  , menuitem
+  , meta
+  , meter
+  , nav
+  , noframes
+  , noscript
+  , object
+  , ol
+  , optgroup
+  , option
+  , output
+  , p
+  , param
+  , pre
+  , progress
+  , q
+  , rp
+  , rt
+  , ruby
+  , s
+  , samp
+  , script
+  , section
+  , select
+  , small
+  , source
+  , span
+  , strong
+  , style
+  , sub
+  , summary
+  , sup
+  , table
+  , tbody
+  , td
+  , textarea
+  , tfoot
+  , th
+  , thead
+  , time
+  , title
+  , tr
+  , track
+  , tt
+  , u
+  , ul
+  , var
+  , video
+  , wbr
+  ) where
+
+import Prelude hiding (map, div, sub)
+import qualified Prelude as Prelude
+import Halogen.HTML.Core (HTML(..), Prop(), TagName(), tagName, element)
+import Halogen.HTML.Properties.Indexed hiding (title)
+import qualified Halogen.HTML.Properties.Indexed as P
+import qualified Halogen.HTML.Elements as E
+
+-- | An array of properties/attributes.
+type Properties r i = Array (IProp (GlobalAttributes r) i)
+
+-- | An HTML element that admits children.
+type Node r p i = Properties r i -> Array (HTML p i) -> HTML p i
+
+-- | An HTML element that does not admit children.
+type Leaf r p i = Properties r i -> HTML p i
+
+erases :: forall r i. Array (IProp r i) -> Array (Prop i)
+erases = Prelude.map erase
+
+a :: forall p i. Node (download :: I, href :: I, hreflang :: I, mediate :: I, rel :: I, target :: I, type :: I) p i
+a = E.a <<< erases
```

Comment:
Perhaps we could use `unsafeCoerce` instead of using the erases no-op? It would mean there's zero overhead at all using `Indexed` over the non-`Indexed` types then:
```
a :: forall p i. Node (download :: I, href :: I, hreflang :: I, mediate :: I, rel :: I, target :: I, type :: I) p i
a = unsafeCoerce E.a
```
I think that should work anyway...

(The compiler isn't clever enough to inline things like `map id` yet)